### ISSUE TITLE
Change in quantity inheritance breaks downstream tools

### ISSUE THREAD

**Issue by keflavich** [astropy/astropy#14514]
This change:
https://github.com/astropy/astropy/blob/36e30a75d3ca704991d6d60d264476022d5b488d/astropy/units/quantity.py#L735

causes a break downstream in spectral-cube, which inherits from `Quantity` for producing projections along different axes.

@mhvk I've seen, recently, that you recommend against inheriting from Quantity, but this has been around for several years now.  Is there any workaround you can suggest?  The choice of `finalize=False` is hard coded at quite a low level here.

The specific failure in spectral-cube is from doing something very simple:
https://github.com/radio-astro-tools/spectral-cube/blob/master/spectral_cube/tests/test_projection.py#L137-L152
basically, when we add two objects of a particular type, we expect their attributes (`_wcs` and `_beam`, for example) to be preserved.

Longer-term, we're looking to replace the structure in spectral-cube with `nddata` inheritance, but that is likely a year-long effort if I'm being extremely optimistic.

**Comment by pllim** [comment]
Looks like this was done in https://github.com/astropy/astropy/pull/14253 cc @mhvk and @taldcroft

**Comment by keflavich** [comment]
I have a simple enough hack-around right now:
https://github.com/radio-astro-tools/spectral-cube/pull/854/commits/e817f8dda6c5b5ad1b334c8a38f69852f0cb42bc
but it took me several hours of confusion to find that.

**Comment by mhvk** [comment]
@keflavich - thanks so much for reporting! Since this is in `-dev` only, we should try to fix it before 5.3 -- I've set the milestone so we don't forget it. If there is no fix, I think we need to revert #14253.

In #14253, the real thing I was trying to avoid is copying `.info`. It seems clear what you need is for `__array_finalize__` to be called, which is reasonable as that generally happens to all `ndarray` subclasses after operations.  But do you care about `.info`? If not, I'll try to think of another way to get the same result...

**Comment by keflavich** [comment]
I don't think we care about `.info`; afaik we don't use it.

Thanks!

**Comment by mhvk** [comment]
@keflavich - what should be a fix is in #14518. Would be wonderful if you could test it...

**Comment by mhvk** [comment]
Actually, reminds me to write a test case... later today...
